monai/apps/auto3dseg/bundle_gen.py

```diff
@@ -0,0 +1,370 @@
+        spec = importlib.util.spec_from_file_location("InferClass", infer_py)
+        infer_class = importlib.util.module_from_spec(spec)
+        sys.modules["InferClass"] = infer_class
+        spec.loader.exec_module(infer_class)
+        return infer_class.InferClass(config_path, *args, **kwargs)
+
+    def predict(self, predict_params=None):
+        """
+        Use the trained model to predict the outputs with a given input image. Path to input image is in the params
+        dict in a form of {"files", ["path_to_image_1", "path_to_image_2"]}. If it is not specified, then the pre-
+        diction will use the test images predefined in the bundle config.
+
+        Args:
+            predict_params: a dict to override the parameters in the bundle config (including the files to predict).
+
+        """
+        if predict_params is None:
+            params = {}
+        else:
+            params = deepcopy(predict_params)
+
+        files = params.pop("files", ".")
+        inferer = self.get_inferer(**params)
+        return [inferer.infer(f) for f in ensure_tuple(files)]
+
+
+# path to download the algo_templates
+algo_zip = "https://github.com/Project-MONAI/MONAI-extra-test-data/releases/download/0.8.1/algo_templates.tar.xz"
+
+md5 = "596c609035595a5d05ee7bdd70511c84"
+
+# default algorithms
+default_algos = {
+    "unet": dict(
+        _target_="unet.scripts.algo.UnetAlgo",
+        template_configs="../algo_templates/unet/configs",
+        scripts_path="../algo_templates/unet/scripts",
+    ),
+    "segresnet2d": dict(
+        _target_="segresnet2d.scripts.algo.Segresnet2DAlgo",
+        template_configs="../algo_templates/segresnet2d/configs",
+        scripts_path="../algo_templates/segresnet2d/scripts",
+    ),
+    "dints": dict(
+        _target_="dints.scripts.algo.DintsAlgo",
+        template_configs="../algo_templates/dints/configs",
+        scripts_path="../algo_templates/dints/scripts",
+    ),
+}
+
+
+class BundleGen(AlgoGen):
+    """
+    This class generates a set of bundles according to the cross-validation folds, each of them can run independently.
+
+    Args:
+        algo: a dictionary that outlines the algorithm to use.
+        data_stats_filename: the path to the data stats file (generated by DataAnalyzer)
+        data_src_cfg_name: the path to the data source file, which contains the data list
+
+    .. code-block:: bash
+
+        python -m monai.apps.auto3dseg BundleGen generate --data_stats_filename="../algorithms/data_stats.yaml"
+    """
+
+    def __init__(self, algos=None, data_stats_filename=None, data_src_cfg_name=None):
+        self.algos = []
+        if algos is None:
+            # trigger the download process
+            sys.path.insert(0, "..")
+            algo_compressed_file = os.path.join("..", "algo_templates.tar")
+            download_and_extract(algo_zip, algo_compressed_file, "..", md5)
+            algos = copy(default_algos)
```

Comment:
perhaps consider putting the downloaded files to a subfolder of the torch hub folder: https://pytorch.org/docs/stable/hub.html?highlight=get_dir#torch.hub.get_dir

I used the directory previously for models, code here:
https://github.com/Project-MONAI/MONAI/blob/f9e8e10fab3a0c0b87708222db96571236c9e764/monai/apps/mmars/mmars.py#L134-L138

so maybe we can create `algo_dir = Path(get_dir()) / "algo_templates"`

Reply:
Yes, the downloaded files need to be managed better. 

My latest commit put the downloaded files to `algo_path/algorithm_templates`, which user can specify `algo_path` or it will be the user's current working directory. 

If we put the to a subfolder of the torch hub, the process will be a bit more invisible (ofc we can log info and let user know too). I personally likes more transparency but don't know which way is better.

```python
class BundleGen(AlgoGen):
    def __init__(self, algo_path: str = ".", algos=None, data_stats_filename=None, data_src_cfg_name=None):
```

Reply:
ok sure, I think user's working directory is better, here I was mainly thinking about caching the original system default values somewhere to avoid repeated downloading..